playground/CMakeLists.txt

```diff
@@ -1,22 +1,8 @@
-PROJECT(playground)
+add_executable( stlab.playground main.cpp )
 
-set(SOURCE
-  main.cpp
-)
+target_link_libraries( stlab.playground
+                       PUBLIC ${Boost_LIBRARIES} 
+                       PUBLIC stlab )
 
-set(HEADERS
-)
-
-add_executable(playground ${SOURCE} ${HEADERS})
-
-include_directories(${Boost_INCLUDE_DIRS})
-
-target_link_libraries(
-    playground
-    ${Boost_LIBRARIES} 
-    ${CMAKE_THREAD_LIBS_INIT} 
-    ${CONAN_LIBS})
-
-
-set_property(TARGET playground PROPERTY CXX_STANDARD 14)
-set_property(TARGET playground PROPERTY CXX_STANDARD_REQUIRED ON)
+set_property( TARGET stlab.playground PROPERTY CXX_STANDARD 14 )
```

Comment:
Commands like:
```
add_compile_option()
include_directories()
link_directories()
link_libraries()
```
operate on the directory level. That means that they propagate downward into child directories (which is great), but not into parent directories. 

If stlab is added as a subdirectory of another project, as a git submodule for instance, the composing project has to duplicate a substantial portion of your cmake script.

The target level properties are different. When an include directory is added to a target---when using the `PUBLIC` or `INTERFACE` keywords, but not the `PRIVATE` keyword---any other target which links to that target transitively picks up those properties.

That's actually why I introduced a `stlab` interface target. By calling 
```
add_subdirectory( stlab/libraries )
target_link_libraries( myLib PUBLIC stlab )
```
myLib is informed of any include directories or compiler flags necessary to use the project
These are great for treating C++ header-only libraries. Unfortunately, currently, there are limitations on what properties can be associated with interface targets, including the `CXX_STANDARD`. 

That said, a directory property is probably more elegant in this case.